parser/v2/scriptparser.go

```diff
@@ -0,0 +1,142 @@
+package parser
+
+import (
+	"strings"
+
+	"github.com/a-h/parse"
+)
+
+var scriptElement = scriptElementParser{}
+
+type scriptElementParser struct{}
+
+func (p scriptElementParser) Parse(pi *parse.Input) (n Node, ok bool, err error) {
```

Comment:
Could be a parseFunc.

Reply:
IIRC, function vars cause an allocation on init, hence a minor preference for not having them.